src/main/java/org/springframework/data/redis/listener/RedisMessageListenerContainer.java

```diff
@@ -689,36 +690,55 @@ private class SubscriptionTask implements SchedulingAwareRunnable {
 		private class PatternSubscriptionTask implements SchedulingAwareRunnable {
 
 			private long WAIT = 500;
-			private long ROUNDS = 3;
+			private AtomicBoolean isThreadRunning = new AtomicBoolean(true);
+
 
 			public boolean isLongLived() {
 				return false;
 			}
 
+			void cancel() {
+				isThreadRunning.set(false);
```

Comment:
this is called exclusively by the `SubscriptionTask` when there is a `RedisConnectionFailureException` or when the container is stopped.